### PR TITLE
Add Troubleshoot the "failed reaching server" error

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?
EDU 124

Adds KB on common error related to certs.

### CODE REVIEW

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+slug: troubleshoot-last-connection-error
+title: Troubleshoot the "failed reaching server" error
+tags:
+  - cloud
+  - namespaces
+date: 2023-05-09T00:00:00Z
+---
+
+The error, `failed reaching server: last connection error`, is usually caused by an expired TLS certification.
```
Comment (flossypurse) [inline kb/how-to-troubleshoot-last-connection-error.md:10]: Could consider flipping the statement:

An expired TLS certification can often cause `failed reaching server: last connection error`

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+In this article, we'll provide a step-by-step guide for developers to troubleshoot and resolve this error, including how to check the certification expiration date, renew the certification, and update the server configuration.
```
Comment (flossypurse) [inline kb/how-to-troubleshoot-last-connection-error.md:12]:
```suggestion
This is a troubleshooting guide that shows you how to do the following:
- Check the certification expiration date
- Renew the certification
- Update the server configuration.
```

Which server? What configuration?

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+By following these steps, you can quickly resolve the following error:
+
+```bash
+failed reaching server: last connection error: connection closed before server preface received
+```
```
Comment (flossypurse) [inline kb/how-to-troubleshoot-last-connection-error.md:18]: Feels super redundant if the first couple of sentences are done well.

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+**Existing certificate management infrastructure**
+
+If you are using an existing certificate management infrastructure, contact the administrator of the infrastructure to renew the certificate.
+
+**Self-signed certificate**
+
+Alternatively, if you are using a self-signed certificate or do not have an existing infrastructure, you can generate a new certificate using OpenSSL or certstrap.
+For information on generating a self-signed certificate, see [Link to this or similar](https://github.com/temporalio/documentation/pull/2024)
+
+### Update the TLS certification in the server configuration
```
Comment (flossypurse) [inline kb/how-to-troubleshoot-last-connection-error.md:56]:
```suggestion
### Update the TLS certification in the server configuration
```

In Cloud?

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+You can do this by contacting the certificate authority (CA) that issued the certificate and requesting a renewal.
+
+**Existing certificate management infrastructure**
+
+If you are using an existing certificate management infrastructure, contact the administrator of the infrastructure to renew the certificate.
+
+**Self-signed certificate**
+
+Alternatively, if you are using a self-signed certificate or don't have an existing infrastructure, you can generate a new certificate using OpenSSL or [certstrap](https://github.com/square/certstrap).
```
Comment (djmagee) [inline kb/how-to-troubleshoot-last-connection-error.md:51]:
```suggestion
If you are using a self-signed certificate or don't have an existing infrastructure, you can generate a new certificate using OpenSSL or [certstrap](https://github.com/square/certstrap).
```

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+### Update the CA certification in the server configuration
+
+Update the new CA certificate in the Temporal Cloud server configuration.
+
+You can update certificates using any of the following methods:
+
+- **[Update certificates using Temporal Cloud UI](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-temporal-cloud-ui)**
+- **[Update certificates using tcld](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-tcld)**
+
+Retry your connection once you've updated the TLS certification in the server configuration.
```
Comment (djmagee) [inline kb/how-to-troubleshoot-last-connection-error.md:64]:
```suggestion
After you update the TLS certification in the server configuration, retry your connection.
```

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+### Set reminders
+
+Don't let your certificates expire.
+Add reminders to your calendar to issue new CA certificates well before the expiration dates of the existing ones.
+
+## Additional resources
+
+You'll troubleshoot the `failed reaching server: last connection error` error caused by an expired TLS certification by following these steps.
```
Comment (djmagee) [inline kb/how-to-troubleshoot-last-connection-error.md:73]:
```suggestion
The preceding steps should help you troubleshoot the `failed reaching server: last connection error` error caused by an expired TLS certificate.
```

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+If this issue persists, verify the Client you are using to connect to the server is using the correct TLS certification, the Client requests reach the server after the roles are fully initialized, or open a ticket at [https://support.temporal.io/](https://support.temporal.io/).
```
Comment (djmagee) [inline kb/how-to-troubleshoot-last-connection-error.md:75]:
```suggestion
If this issue persists, verify that the Client you are using to connect to the server is using the correct TLS certification and that the Client requests reach the server after the roles are fully initialized.
If you still need help, [create a support ticket](/cloud/how-to-create-a-ticket-for-temporal-support).
```

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+For information on generating a self-signed certificate, see [Control authorization](/cloud/how-to-manage-certificates-in-temporal-cloud#control-authorization).
+
+### Update the CA certification in the server configuration
+
+Update the new CA certificate in the Temporal Cloud server configuration.
+
+You can update certificates using any of the following methods:
+
+- **[Update certificates using Temporal Cloud UI](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-temporal-cloud-ui)**
+- **[Update certificates using tcld](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-tcld)**
```
Comment (djmagee) [inline kb/how-to-troubleshoot-last-connection-error.md:62]:
```suggestion
- [Update certificates using Temporal Cloud UI](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-temporal-cloud-ui)
- [Update certificates using tcld](https://docs.temporal.io/cloud/how-to-manage-certificates-in-temporal-cloud#update-certificates-using-tcld)
```

**Review by flossypurse — APPROVED** [review]
Stamping so that once you get SME approval and get an editorial approval you are gtg.

File: kb/how-to-troubleshoot-last-connection-error.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+If you are using a self-signed certificate, run the following `tctl` command:
+
+```command
+tctl --namespace <namespace_id> --address <address_id>.tmprl.cloud:7233 --tls-cert-path ~/certs/path.pem --tls-key-path ~/certs/path.key namespace describe
+```
+
+### Renew TLS certification
+
+If the certificate has expired or is about to expire, the next step is to renew it.
+
```
Comment (antmendoza) [inline kb/how-to-troubleshoot-last-connection-error.md:44]: with tcld customer can list their certificates and the expiration date, not sure if it helps 
e.g `tcld namespace accepted-client-ca list --namespace antonio.a2dd6`

**Pull request merged**